gtfslib/converter.py

```diff
@@ -411,21 +412,38 @@ def import_stop(stop, stoptype, zone_ids, item_ids, station_ids=None):
         lat = _tofloat(shpt.get('shape_pt_lat'))
         lon = _tofloat(shpt.get('shape_pt_lon'))
         n_shape_pts += 1
+        shape_flush_pt_counter += 1
         if n_shape_pts % 10000 == 0:
             logger.info("%d shape points" % n_shape_pts)
-            dao.flush()
+
         shape = shapes.get(shape_id)
         if shape is None:
+            if shape_flush_pt_counter > 100000:
+                
+                dao.bulk_save_objects(shapes.values())
+                for sk in shapes:
+                    dao.bulk_save_objects(shapes[sk].points)                
+                dao.flush()
+                logger.info('%s shapes and %s points inserted and flushed' % (len(shapes),shape_flush_pt_counter))
+                shapes = {}
+                shape_flush_pt_counter-=100000
+
```

Comment:
Unless I'm mistaken, but this assume shape points to be in order of shape in the input data, which is not necessarily the case for all GTFS. Is it necessary to clear the `shapes` set here? I guess we could only flush shape points and leave all shapes in memory (a shape should be really just an ID so should not take that much memory).